### ISSUE TITLE
Could not identify a smart meter on your account with gateway access

### ISSUE THREAD

**Issue by slappula** [mjmeli/ha-duke-energy-gateway#10]
I'm receiving this message in the logs:

Could not identify a smart meter on your account with gateway access

Any idea on how to address this one? Version 0.0.6. Details from log:

_This error originated from a custom integration.

Logger: custom_components.duke_energy_gateway
Source: custom_components/duke_energy_gateway/__init__.py:82 
Integration: Duke Energy Gateway (documentation, issues) 
First occurred: 10:09:06 AM (2 occurrences) 
Last logged: 10:11:53 AM

Could not identify a smart meter on your account with gateway access._

**Comment by mjmeli** [comment]
I would guess there is some difference in your API responses from mine that is causing it to get filtered out. The code tries to filter down to supported meters but the logic could be bad
https://github.com/mjmeli/ha-duke-energy-gateway/blob/e19239f2532217170b5da6bed3ed8a017ec84b26/custom_components/duke_energy_gateway/__init__.py#L53-L74

A couple of ideas to help track it down

1. Enable debug logging for the integration by adding below to your configuration.yaml. Then in your full home-assistant.log file you will see some logs tagged custom_components.duke_energy_gateway. Unfortunately it might not give much insight because it only logs happy path right now.
```
logger:
  default: info
  logs:
    custom_components.duke_energy_gateway: debug
```
2. If you are able to, you could run the `example.py` script in the API library (https://github.com/mjmeli/pyduke-energy/blob/main/example.py). This will dump out the API payloads so it might be more straightforward and insightful.

**Comment by slappula** [comment]
Thanks for the quick reply! Also, I forgot to say earlier- thanks for developing this!

I turned on the logging and I see this:

2021-08-23 14:15:22 DEBUG (MainThread) [custom_components.duke_energy_gateway] Setup Duke Energy API client
2021-08-23 14:15:23 DEBUG (MainThread) [custom_components.duke_energy_gateway] Checking account 9100xxxxxx for gateway
2021-08-23 14:15:27 DEBUG (MainThread) [custom_components.duke_energy_gateway] Failed to find meter on account 910021843313: Request failed with unexpected error [https://cust-api.duke-energy.com/gep/v2/auth/account-details]: 500, message='Internal Server Error', url=URL('https://cust-api.duke-energy.com/gep/v2/auth/account-details?email=dummy@dummy.com&srcSysCd=ISU&srcAcctId=9100xxxxxx')
2021-08-23 14:15:27 ERROR (MainThread) [custom_components.duke_energy_gateway] Could not identify a smart meter on your account with gateway access.

My Python install is borked (or I'm not running something right). Either way- I can't get example.py to run. I tried to use the URI from the logs and run it via Postman but I'm getting an authentication error. Is there a way to do that? If yes, what auth method would I use that would allow me to plug in my email address/password?

**Comment by mjmeli** [comment]
At first glance the only difference I notice is that your srcSysCd value is different. I'm not sure what that actually is but mine is "CIM" and yours is "ISU".

And just to make sure, that account number matches the account number on the account with the gateway?

You can definitely query these via Postman.

To do authentication, you need to POST to `https://cust-api.duke-energy.com/gep/v2/auth/oauth2/token` with header `Authorization: Basic UzdmNXFQR2MwcnpVZkJmcVNPak9ycGczZWtSZ3ZHSng6bW1nS2pyY1RQRHptOERtVw==` (this is a hard-coded value in the app) and a request body as x-www-form-encoded of grant_type=password, username={email}, and password={password}.

That will return an access_token which you use for the other call with as a Bearer token. You might also save the cdp_internal_user_id value.

To get your sysSysCd value, you can GET to `https://cust-api.duke-energy.com/gep/v2/auth/account-list` using that access_token and parameters email={email} and internalUserId={cdp_internal_user_id}

Maybe there is something in your responses that gives insight into what is different between us.

**Comment by amancyh** [comment]
Thanks for the integration; mine is giving error "failed to set up"

This error originated from a custom integration.

Logger: custom_components.duke_energy_gateway
Source: custom_components/duke_energy_gateway/__init__.py:82
Integration: Duke Energy Gateway (documentation, issues)
First occurred: 5:07:43 PM (1 occurrences)
Last logged: 5:07:43 PM

Could not identify a smart meter on your account with gateway access.

**Comment by mjmeli** [comment]
I believe this is fixed in the newest release 0.0.7. I've also added some more logging in the event we need to debug this again in the future.

Try updating in HACS and let me know if there are still issues

**Comment by slappula** [comment]
Seems to be working great after updating to 0.0.7. Thanks again!
